maven-plugin/src/main/java/org/openmrs/maven/plugins/BuildDistro.java

```diff
@@ -129,14 +137,43 @@ public void executeTask() throws MojoExecutionException, MojoFailureException {
 		if (distroProperties == null) {
 			Server server = new Server.ServerBuilder().build();
 
-			wizard.promptForRefAppVersionIfMissing(server, versionsHelper, DISTRIBUTION_VERSION_PROMPT);
-			if (DistroHelper.isRefapp2_3_1orLower(server.getDistroArtifactId(), server.getVersion())) {
-				distroProperties = new DistroProperties(server.getVersion());
-			} else {
-				distroProperties = distroHelper.downloadDistroProperties(buildDirectory, server);
-				distroArtifact = new Artifact(server.getDistroArtifactId(), server.getVersion(), server.getDistroGroupId(),
-						"jar");
+			List<String> options = new ArrayList<>();
+			options.add(O2_DISTRIBUTION);
+			options.add(O3_DISTRIBUTION);
+
+			String choice = wizard.promptForMissingValueWithOptions("You can setup following servers", null, null, options);
+			switch (choice) {
+				case O2_DISTRIBUTION:
+					wizard.promptForRefAppVersionIfMissing(server, versionsHelper, DISTRIBUTION_VERSION_PROMPT);
+					if (DistroHelper.isRefapp2_3_1orLower(server.getDistroArtifactId(), server.getVersion())) {
+						distroProperties = new DistroProperties(server.getVersion());
+					} else {
+						distroProperties = distroHelper.downloadDistroProperties(buildDirectory, server);
+						distroArtifact = new Artifact(server.getDistroArtifactId(), server.getVersion(), server.getDistroGroupId(),
+								"jar");
+					}
+					break;
+				case O3_DISTRIBUTION:
+					wizard.promptForO3RefAppVersionIfMissing(server, versionsHelper);
```

Comment:
What you are doing for O3 looks more than what the `build-distro` command is supposed to be doing. For instance, why should i need docker to be running for this command to execute successfully?